### PR TITLE
crypto: Fix Kconfig dependencies and cleanup 

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Fix Kconfig dependencies b/w ext/lib/crypto and drivers/crypto 

Minor clean up changes drivers/crypto

### CODE REVIEW

File: drivers/crypto/crypto_ataes132a.c [hunk 1]
```diff
@@ -779,7 +779,8 @@ static int do_block(struct cipher_ctx *ctx, struct cipher_pkt *pkt)
 	return ataes132a_aes_ecb_block(dev, key_id, pkt);
 }
 
-int ataes132a_session_free(struct device *dev, struct cipher_ctx *session)
+static int ataes132a_session_free(struct device *dev,
+				struct cipher_ctx *session)
```
Comment (tbursztyka) [inline drivers/crypto/crypto_ataes132a.c]: align the parameter to the last '(' + 1 space.

Apply that everywhere.

**Review by tbursztyka — COMMENTED** [review]
Just a tiny comment. 

Rest looks just fine to me.

File: drivers/crypto/crypto_ataes132a.c (hunk 1 above)
Comment (ramakrishnapallala) [reply drivers/crypto/crypto_ataes132a.c]: ok got it.

**Comment by andrewboie** [comment]
please fix commit message

> Commit f4b0ee498b:
1: UC5 Title exceeds max length (73>72): "drivers: crypto: Limit the scope of ataes132a crypto_driver_api functions"

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by pfalcon** [comment]
> Set default MBEDTLS_HEAP_SIZE to 512

I wonder if this is a truly good default, or will just land-lock 0.5K of RAM for apps which don't use full TLS, but only simple crypto algos.

**Comment by ramakrishnapallala** [comment]
@pfalcon , i have set the minimum value from the sample apps settings.
Anyways applications can always change this default value.

**Pull request merged**